brd-ios/breadwallet/src/Redesign/Scenes/Swap/Details/SwapDetailsPresenter.swift

```diff
@@ -16,89 +16,169 @@ final class SwapDetailsPresenter: NSObject, Presenter, SwapDetailsActionResponse
     // MARK: - SwapDetailsActionResponses
     func presentData(actionResponse: FetchModels.Get.ActionResponse) {
         guard let item = actionResponse.item as? Models.Item else { return }
+        let detail = item.detail
+        let type = item.type
         
-        let sections = [
-            Models.Section.header,
-            Models.Section.order,
-            Models.Section.fromCurrency,
-            Models.Section.image,
-            Models.Section.toCurrency,
-            Models.Section.timestamp,
-            Models.Section.transactionFrom,
-            Models.Section.transactionTo
-        ]
+        var sections = [AnyHashable]()
+        
+        switch type {
+        case .swapTransaction:
+            sections = [
+                Models.Section.header,
+                Models.Section.order,
+                Models.Section.fromCurrency,
+                Models.Section.image,
+                Models.Section.toCurrency,
+                Models.Section.timestamp,
+                Models.Section.transactionFrom,
+                Models.Section.transactionTo
+            ]
+            
+        case .buyTransaction:
+            sections = [
+                Models.Section.header,
+                Models.Section.toCurrency,
+                Models.Section.buyOrder,
+                Models.Section.order,
+                Models.Section.timestamp,
+                Models.Section.transactionTo
+            ]
+            
+        default:
+            break
+        }
```

Comment:
![Uploading Screenshot 2022-08-30 at 08.04.11.png…]()

Reply:
The link for screenshot is not working :/